src/libsync/owncloudpropagator.cpp

```diff
@@ -618,30 +623,31 @@ bool PropagateDirectory::scheduleNextJob()
         return false;
     }
 
-    // cache the value of first unfinished subjob
     bool stopAtDirectory = false;
-    int i = _firstUnfinishedSubJob;
-    int subJobsCount = _subJobs.count();
-    while (i < subJobsCount && _subJobs.at(i)->_state == Finished) {
-      _firstUnfinishedSubJob = ++i;
-    }
```

Comment:
This will fix this sophisticated caching. Is there any reason to keep Finished Jobs?